apis/v1alpha1/httproute_types.go

```diff
@@ -165,32 +180,68 @@ type HTTPRouteMatch struct {
 	ExtensionRef *RouteMatchExtensionObjectReference `json:"extensionRef" protobuf:"bytes,5,opt,name=extensionRef"`
 }
 
-// HTTPRouteFilter defines a filter-like action to be applied to
-// requests.
-type HTTPRouteFilter struct {
-	// Headers related filters.
-	//
-	// Support: extended
+const (
+	// FilterHTTPReqeustHeaderTransformer can be used to add or remove an HTTP
+	// header from an HTTP request before it is sent to the upstream target.
+	// Support: core
 	// +optional
-	Headers *HTTPHeaderFilter `json:"headers" protobuf:"bytes,1,opt,name=headers"`
+	FilterHTTPReqeustHeaderTransformer = "HTTPRequestHeaderTransformer"
+	// TODO(hbagdi): add FilterHTTPRequestMirroring filter
+)
+
+// HTTPRouteFilter defines additional processing steps that must be completed
+// during the request or response lifecycle.
+// HTTPRouteFilters are meant as an extension point to express additional
+// processing that must be done in Gateway implementations. Some examples include
+// request or response modification, implementing authentication strategies,
+// rate-limiting, and traffic shaping.
+// API guarantee/conformance is defined based on the type of the filter.
+// +union
+type HTTPRouteFilter struct {
+	// Type identifies the filter to execute.
+	// Types are classified into two conformance-levels:
+	// - Core and extended: These filter types and their corresponding configuration
+	//   is defined in this package. All implementations must implement
+	//   the core filters. Implementers are encouraged to support extended filters.
+	//   Definitions for filter-specific configuration for these
+	//   filters is defined in this package.
+	// - Custom: These filters are defined and supported by specific vendors.
+	//   In future, filters showing convergence in behavior across multiple
+	//   implementations will be considered for inclusion in extended or core
+	//   conformance rings. Filter-specific configuration can be specified
+	//   using ExtensionRef or Config fields.
+	// Implementers are encouraged to define custom implementation
+	// types to extend the core API with implementation-specific behavior.
+	//
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	// +required
```

Comment:
I don't believe we use the `+required` tag with other required fields. It's preferable that we are consistent across the api's.

Reply:
I might be misunderstand you but we use `+required` in a number of places gateway_types.go. :thinking:

Reply:
Do we get any mileage from the CRD validation with required? This sounds like something that can be done in bulk through the whole repo.

Reply:
> Do we get any mileage from the CRD validation with required?

No. Anything without `+optional` tag is marked as required in CRD validation.

Reply:
But `+required` is nice and obvious :)